subsys/bluetooth/host/att.c

```diff
@@ -3088,9 +3105,40 @@ int bt_eatt_connect(struct bt_conn *conn, uint8_t num_channels)
 		return -ENOMEM;
 	}
 
-	return bt_l2cap_ecred_chan_connect(conn, chan, BT_EATT_PSM);
+	while (offset < i) {
+		/* bt_l2cap_ecred_chan_connect() uses the first L2CAP_ECRED_CHAN_MAX_PER_REQ
+		 * elements of the array or until a null-terminator is reached.
+		 */
+		err = bt_l2cap_ecred_chan_connect(conn, &chan[offset], BT_EATT_PSM);
+		if (err < 0) {
+			return err;
+		}
+
+		offset += L2CAP_ECRED_CHAN_MAX_PER_REQ;
+	}
+
+	return 0;
 }
 
+#if defined(CONFIG_BT_EATT_AUTO_CONNECT)
+void eatt_auto_connect(struct bt_conn *conn, uint8_t err)
+{
+	int eatt_err;
+
```

Comment:
Should this be checking if the connection is actually a central or not? I know you probably want to test if the collision handling works but I don't think it is a good idea to always start connecting like this, also have you tried connecting with something like BlueZ that supports EATT as well and would start connecting immediately as central?

Reply:
This is done by `att_schedule_eatt_connect` which adds a delay for peripherals based on `credit_based_connection_delay`